Sri Hari: Batch-2/Neetcode-150/articles

Srihari2222 opened the pull request:
@neetcode-gh 

- Took 10 hours for this work.
- Please review it.  

---
1. There is a problem in python environment on neetcode.io for the problem [coin-change-ii](https://neetcode.io/problems/coin-change-ii).
2. The memorized solution is getting error in case of python and is working fine for all other langueages.
3. Please check that.
![Screenshot 2024-10-20 104642](https://github.com/user-attachments/assets/3fc7a412-77d6-401c-80e8-99b191b82251)
![Screenshot 2024-10-20 104717](https://github.com/user-attachments/assets/fccc0488-bcef-4394-a2d2-6827292a94e6)

neetcode-gh commented:
@Srihari2222 Thanks for letting me know, this is fixed.

Minor: the partition equal subset article has solutions numbered as the 5th solution.

It looks like you made some naming changes to previous files, but can you describe the changes in the sliding-window-maximum file? It looks like for the JS deque solution you reverted some of the changes I made. Perhaps you forgot to pull them?

Btw it seems like each PR is taking more hours then previously, is there any issues you're encountering?

Srihari2222 commented:
@neetcode-gh 

- Sorry for the wrong changes made.
- That was a typo, 

---
### Changes in sliding window problem
1. Modified the segment tree code as previous code was very inefficient. I mean the segment tree initialization was very bad.
2. Ok, that was my mistake regarding deque. Modified it.

--- 
I was writing this PR and the problems were regarding 2D - dynamic programming.
The main problem I faced about that coin change-2 python error. That article took my maximum time.
Sorry, I am trying my best regarding the time constraint you mentioned. It will not repeat again.

neetcode-gh commented:
@Srihari2222 No problem at all, it's not a big deal. This is just part of the learning process. You're doing a great job.

And thanks for letting me know, if you come accross and test cases that could be added to NC feel free to send them to me in a separate email.
